File: backend/src/lista/lista.controller.ts
import { Controller, Post, Body, Get, Put, Delete, Param } from '@nestjs/common';
import { ListaService } from './lista.service';
import { ListaTarefa } from './lista.entity';

@Controller('listaTarefa')
export class ListaController {

  constructor(private service: ListaService) { }

  @Post()
  async criarTarefa(@Body('descricao') descricao: string) {
    return await this.service.save(descricao);
  }

  @Get()
  async listarTarefa() {
    return await this.service.findAll();
  }

  @Put()
  async atualizarTarefa(@Body() lista: ListaTarefa) {
    return await this.service.update(lista);
  }

  @Delete(':cod')
  async removerTarefa(@Param('cod') cod: number) {
    return await this.service.remove(cod);
  }

}

File: mobile/lib/listaDeTarefa/lista.controller.dart
import 'package:ListaTarefa/listaDeTarefa/lista.model.dart';
import 'package:ListaTarefa/listaDeTarefa/lista.service.dart';
import 'package:mobx/mobx.dart';

part 'lista.controller.g.dart';

class ListaController = _ListaControllerBase with _$ListaController;

abstract class _ListaControllerBase with Store {
  @observable
  List<ListaTarefa> listaTarefas = [];

  @action
  fetchLista() async {
    listaTarefas = await ListaService().listAll();
  }

  @action
  create(String descricao) async {
    await ListaService().create(descricao);
    fetchLista();
  }

  @action
  remove(int cod) async {
    await ListaService().remove(cod);
    fetchLista();
  }

  @action
  update(ListaTarefa listaTarefa) async {
    await ListaService().update(listaTarefa);
    fetchLista();
  }
}
